src/plugins/htmlVideoPlayer/plugin.js

```diff
@@ -265,13 +265,53 @@ function tryRemoveElement(elem) {
         #resizeObserver;
 
         constructor() {
+            this.onWindowResize = this.onWindowResize.bind(this);
             if (browser.edgeUwp) {
                 this.name = 'Windows Video Player';
             } else {
                 this.name = 'Html Video Player';
             }
         }
 
+        bindEvents() {
+            window.addEventListener('resize', this.onWindowResize);
+        }
+
+        unbindEvents() {
+            window.removeEventListener('resize', this.onWindowResize);
+        }
+
+        onWindowResize() {
+            if (!this.#mediaElement
+                || !this.#mediaElement.videoWidth
+                || !this.#mediaElement.videoHeight) return;
+            const subtitleCanvas = this.#currentSubtitlesOctopus
+                ? document.querySelector('.libassjs-canvas')
+                : undefined;
+            switch (this._currentAspectRatio) {
+                case 'fill': {
+                    this.#mediaElement.setAttribute('style', 'object-fit: fill');
+                    if (subtitleCanvas) subtitleCanvas.style.removeProperty('margin-left');
+                    break;
+                }
+                case 'cover': {
+                    const coverWidth = Math.round(window.innerHeight / this.#mediaElement.videoHeight * this.#mediaElement.videoWidth);
+                    const centerLeft = Math.round((coverWidth - window.innerWidth) / 2);
+                    this.#mediaElement.setAttribute('style', `margin-left: -${centerLeft}px !important; width: ${coverWidth}px`);
+                    if (subtitleCanvas) subtitleCanvas.style.marginLeft = `-${centerLeft}px`;
+                    break;
+                }
+                default: {
+                    this.#mediaElement.removeAttribute('style');
+                    if (subtitleCanvas) subtitleCanvas.style.removeProperty('margin-left');
+                    break;
+                }
+            }
+            if (this.#currentSubtitlesOctopus) {
+                this.#currentSubtitlesOctopus.resetRenderAheadCache(false);
```

Comment:
**4K TV issue**
On Tizen 4 [there is no `ResizeObserver`](https://github.com/libass/JavascriptSubtitlesOctopus/blob/1e9d5cd2147420fe1485bd1b71fbc9ee51537641/src/subtitles-octopus.js#L200-L204), so JSO resize handler is not called.
Fixed by adding `if (typeof ResizeObserver === 'undefined') this.#currentSubtitlesOctopus.resize();` here.